nixos/modules/services/networking/firewall.nix

```diff
@@ -179,17 +179,19 @@ let
 
     # Accept packets on the extra allowed host+port combination.
     ${concatMapStrings (allowed:
-      if (allowed.useIPv4 || allowed.usedIPv6) then (
-        (if allowed.useIPv4 then (
-           if allowed.useIPv6 then ("ip46tables -A nixos-fw ") else ("iptables -A nixos-fw ")
-        ) else (
-           "ip6tables -A nixos-fw "
-        ))
-        + (if (allowed.source != "") then (" --source " + allowed.source) else "")
-        + (if (allowed.protocol != "") then (" -p " + allowed.protocol) else "")
-        + (if (allowed.dport != 0) then (" --dport " + toString allowed.dport) else "")
-        + " -j nixos-fw-accept\n"
-      ) else "") cfg.extraAllowed
+        concatMapStrings (addressFamily:
+        concatMapStrings (protocol:
+          (if addressFamily == "IPv4" then "iptables -A nixos-fw "
+           else "ip6tables -A nixos-fw ")
+          + (if allowed.sourceAddress != null then (" --source " + allowed.sourceAddress) else "")
```

Comment:
Using `lib.optionalString` to conditionally interpolate is best practice in nixpkgs.